pkgs/by-name/ha/hare/package.nix

```
@@ -0,0 +1,117 @@
+{ lib
+, stdenv
+, fetchFromSourcehut
+, binutils-unwrapped
+, harec
+, makeWrapper
+, qbe
+, scdoc
+, tzdata
+, substituteAll
+}:
+
+let
+  # We use harec's override of qbe until 1.2 is released, but the `qbe` argument
+  # is kept to avoid breakage.
+  qbe = harec.qbeUnstable;
+  # Until #265124 is merged (it's already on staging), this override is needed.
+  tzdata' = tzdata.overrideAttrs (old: {
+    postInstall = old.postInstall + ''
+      cp leap-seconds.list $out/share/zoneinfo
+    '';
+  });
+  # https://harelang.org/platforms/
+  arch =
+    if stdenv.hostPlatform.isx86_64 then "x86_64"
+    else if stdenv.hostPlatform.isAarch64 then "aarch64"
+    else if stdenv.hostPlatform.isRiscV64 then "riscv64"
+    else builtins.throw "Unsupported architecture: ${stdenv.hostPlatform.system}";
+  platform =
+    if stdenv.hostPlatform.isLinux then "linux"
+    else if stdenv.hostPlatform.isFreeBSD then "freebsd"
+    else builtins.throw "Unsupported platform: ${stdenv.hostPlatform.system}";
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "hare";
+  version = "unstable-2023-11-27";
+
+  outputs = [ "out" "man" ];
+
+  src = fetchFromSourcehut {
+    owner = "~sircmpwn";
+    repo = "hare";
+    rev = "d94f355481a320fb2aec13ef62cb3bfe2416f5e4";
+    hash = "sha256-Mpl3VO4xvLCKHeYr/FPuS6jl8CkyeqDz18mQ6Zv05oc=";
+  };
+
+  patches = [
+    # Replace FHS paths with nix store
+    (substituteAll {
+      src = ./001-tzdata.patch;
+      tzdata = tzdata';
+    })
+  ];
+
+  nativeBuildInputs = [
+    binutils-unwrapped
+    harec
+    makeWrapper
+    qbe
+    scdoc
+  ];
+
+  buildInputs = [
+    binutils-unwrapped
+    harec
+    qbe
+    tzdata'
+  ];
+
+  makeFlags = [
+    "HARECACHE=.cache"
+    "PREFIX=${builtins.placeholder "out"}"
+    "PLATFORM=${platform}"
+    "ARCH=${arch}"
+  ];
+
+  # Append the distribution name to the version
+  env.LOCALVER = "nix";
+
+  doCheck = true;
+
+  strictDeps = true;
+
+  enableParallelBuilding = true;
+
+  configurePhase = ''
+    runHook preConfigure
+
+    ln -s config.example.mk config.mk
+
+    runHook postConfigure
+  '';
+
+  postFixup =
+    let
+      binPath = lib.makeBinPath [
+        binutils-unwrapped
+        harec
+        qbe
+      ];
+    in
+    ''
+      wrapProgram $out/bin/hare --prefix PATH : ${binPath}
+    '';
```

```suggestion
  postFixup = ''
    wrapProgram $out/bin/hare \
      --prefix PATH : ${lib.makeBinPath [ binutils-unwrapped harec qbe ]}
  '';
```
lets compact this a bit